pkg/dockerregistry/server/pullthroughmanifestservice_test.go

```diff
@@ -704,3 +711,74 @@ func TestPullthroughManifestMirroring(t *testing.T) {
 		t.Fatal("timeout while waiting for manifest to be mirrored")
 	}
 }
+
+func TestPullthroughManifestMetrics(t *testing.T) {
+	namespace := "myproject"
+	repo := "myapp"
+	repoName := fmt.Sprintf("%s/%s", namespace, repo)
+
+	mediaType := "application/vnd.docker.distribution.manifest.v2+json"
+	manifest := `{"schemaVersion":2,"mediaType":"` + mediaType + `"}`
+	config := `{}`
+
+	manifestDigest := digest.FromBytes([]byte(manifest))
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/v2/":
+			fmt.Fprint(w, "{}")
+		case "/v2/remoteimage/manifests/" + manifestDigest.String():
+			w.Header().Set("Content-Type", mediaType)
+			fmt.Fprint(w, manifest)
+		default:
+			t.Logf("unhandled request: %s %v", r.Method, r.URL)
+			http.Error(w, "404 not found", http.StatusNotFound)
+		}
+	}))
+	defer ts.Close()
+
+	tsURL, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatalf("failed to parse test server URL: %v", err)
+	}
+
+	img, err := testutil.NewImageForManifest("unused", manifest, config, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	img.DockerImageReference = fmt.Sprintf("%s/remoteimage", tsURL.Host)
+	img.DockerImageManifest = ""
+	img.DockerImageConfig = ""
+
+	ctx := context.Background()
+	ctx = testutil.WithTestLogger(ctx, t)
+
+	fos, imageClient := testutil.NewFakeOpenShiftWithClient(ctx)
+	testutil.AddImageStream(t, fos, namespace, repo, map[string]string{
+		imageapi.InsecureRepositoryAnnotation: "true",
+	})
+	testutil.AddImage(t, fos, img, namespace, repo, "latest")
+
+	imageStream := imagestream.New(ctx, namespace, repo, registryclient.NewFakeRegistryAPIClient(nil, imageClient))
+
+	c, sink := metricstesting.NewCounterSink()
+	ms := newTestManifestService(repoName, nil)
+	ptms := &pullthroughManifestService{
+		ManifestService:         ms,
+		newLocalManifestService: func(ctx context.Context) (distribution.ManifestService, error) { return ms, nil },
+		imageStream:             imageStream,
+		metrics:                 metrics.NewMetrics(sink),
+	}
+
+	_, err = ptms.Get(ctx, digest.Digest(img.Name))
+	if err != nil {
+		t.Fatalf("failed to get manifest: %v", err)
+	}
+
+	if diff := c.Diff(counter.M{
```

Comment:
The test could be repeated for 404 case as well to give us a grasp of metric's behaviour in error cases.